simulator/func_sim/func_sim.cpp

```diff
@@ -56,23 +56,60 @@ typename FuncSim<ISA>::FuncInstr FuncSim<ISA>::step()
 }
 
 template <typename ISA>
-void FuncSim<ISA>::init( const std::string& tr)
+Simulator::StopReason FuncSim<ISA>::run( uint64 instrs_to_run)
 {
+    if (!binary_file_loaded)
+        throw NoBinaryFile();
+    for ( uint64 i = 0; i < instrs_to_run; ++i) {
+        const auto& instr = step();
+        sout << instr << std::endl;
+
+        if ( instrs_to_run == 1)
```

Comment:
Does not look good. What if I want to run a single instruction and it is a halt or `break`?
I think that stepping has to be a separate function, as its action is different to continuous running.